src/main/java/ru/investbook/LoadingPageServer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * InvestBook
+ * Copyright (C) 2024  Spacious Team <spacious-team@ya.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package ru.investbook;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+
+import static java.nio.charset.StandardCharsets.*;
+
+@Slf4j
+public class LoadingPageServer {
+    public static final int SERVER_PORT = 2031;
+
+    private HttpServer server;
+
+    public void start() {
+        try {
+            server = HttpServer.create(new InetSocketAddress(SERVER_PORT), 0);
```

Comment:
You found a good solution for port read from property file. Would you, please, read here `investbook.open-home-page-after-start` property value from relative path file `./application-conf.properties`. If you create the Helper class for reading properties and reuse it for `PortHandler`, it will be the best choise.
```java
try {
    if (!shouldOpenHomePageAfterStart()) {
        return;
    }
    ....
```
You may use static import for Helper's class method.